src/js/polyfills/console.js

```diff
@@ -1,258 +1,316 @@
+                    if (typeof data[i] === 'object') {
+                        for (const key in data[i]) {
+                            if (!propsS.has(key)) {
+                                props.push(key);
+                                propsS.add(key);
+                            }
+                        }
+                    }
+                }
 
-    return out;
-}
+                return props;
+            }
 
-function cliTable(head, columns) {
-    const rows = [];
-    const columnWidths = head.map(h => countBytes(h));
-    const longestColumn = columns.reduce(
-        (n, a) => Math.max(n, a.length),
-        0
-    );
+            if (!properties) {
+                properties = getProperties(data);
+            }
 
-    for (let i = 0; i < head.length; i++) {
-        const column = columns[i];
+            function normalize(data) {
+                // eslint-disable-next-line no-control-regex
+                const colorRegExp = /\u001b\[\d\d?m/g;
 
-        for (let j = 0; j < longestColumn; j++) {
-            if (rows[j] === undefined) {
-                rows[j] = [];
+                return inspect(data).replace(colorRegExp, '');
             }
 
-            const value = (rows[j][i] = hasOwnProperty(column, j) ? column[j] : '');
-            const width = columnWidths[i] || 0;
-            const counted = countBytes(value);
-
-            columnWidths[i] = Math.max(width, counted);
-        }
-    }
+            function countBytes(str) {
+                return encoder.encode(str).byteLength;
+            }
 
-    const divider = columnWidths.map(i => tableChars.middleMiddle.repeat(i + 2));
+            function getTableData(data, properties, addIndex = true) {
+                const rows = [
+                    addIndex ? [ '(index)', ...properties ] : [ ...properties ]
+                ];
 
-    let result =
-    `${tableChars.topLeft}${divider.join(tableChars.topMiddle)}` +
-    `${tableChars.topRight}\n${renderRow(head, columnWidths)}\n` +
-    `${tableChars.leftMiddle}${divider.join(tableChars.rowMiddle)}` +
-    `${tableChars.rightMiddle}\n`;
+                for (const i in data) {
+                    const row = addIndex ? [ i ] : [];
 
-    for (const row of rows) {
-        result += `${renderRow(row, columnWidths)}\n`;
-    }
+                    for (const p of properties) {
+                        row.push(normalize(data[i][p] || ''));
+                    }
 
-    result +=
-    `${tableChars.bottomLeft}${divider.join(tableChars.bottomMiddle)}` +
-    tableChars.bottomRight;
+                    rows.push(row);
+                }
 
-    return result;
-}
+                return rows;
+            }
 
+            const rows = getTableData(data, properties);
 
-class Console {
-    log(...args) {
-        print(...args);
-    }
+            const cols = [];
 
-    info(...args) {
-        print(...args);
-    }
+            for (let ci=0;ci<rows[0].length;ci++) {
```

Comment:
style: please add spacing

Reply:
Not sure where exactly you meant. Maybe got lost during force-push. But eslint doesn't complain.